### PR TITLE
clarifying verification logic

### PR DESCRIPTION
**Pull request by pdg744**
- Porting bolton's comments from risczero-wip.
- @jbruestle there are a couple of TODOs here about parts of the verify logic that are still somewhat opaque to me -- hoping to clarify those before landing this.

### CODE REVIEW

**Review by BoltonBailey — APPROVED** [review]

File: risc0/zkp/src/verify/mod.rs [hunk 1]
```diff
@@ -347,28 +353,42 @@ where
...
+    // Get a pseudorandom DEEP query point
+    // See DEEP-ALI protocol from DEEP-FRI paper for details on DEEP query.
     let z = H::ExtElem::random(&mut iop);
     // debug!("Z = {z:?}");
     let back_one = <H::Elem as RootsOfUnity>::ROU_REV[po2 as usize];
 
-    // Read the U coeffs + commit their hash
+    // Read the U coeffs (the coefficients of the DEEP polynomial) + commit their
+    // hash
+    // TODO - are U coeffs the interpolations of the taps or the DEEP quotient poly?
```
Comment (jbruestle) [inline risc0/zkp/src/verify/mod.rs:385]: Interpolations of the taps.

File: risc0/zkp/src/verify/mod.rs [hunk 2]
```diff
@@ -382,6 +402,7 @@ where
     assert_eq!(eval_u.len(), num_taps, "Miscalculated capacity for eval_us");
 
     // Compute the core polynomial
+    // TODO - clarify: what is the "core polynomial?"
```
Comment (jbruestle) [inline risc0/zkp/src/verify/mod.rs:405]: The constraint polynomial.  I.e. the set of all constraints mixed by poly_mix

File: risc0/zkp/src/verify/mod.rs [hunk 3]
```diff
@@ -424,12 +445,13 @@ where
...
     }
 
-    // Set the mix mix value
+    // Set the mix mix value, pseudorandom value used for FRI batching
     let mix = H::ExtElem::random(&mut iop);
     // debug!("mix = {mix:?}");
 
     // Make the mixed U polynomials.  combo_u has one element for each
     // back for each combo, plus 1 for the check.
+    // TODO what is combo_u?
```
Comment (jbruestle) [inline risc0/zkp/src/verify/mod.rs:454]: So if you look at the DEEP-ALI equation, you can see that the denominator is shared for all columns which have the same set of taps.  If you group all terms with the same denominator, you can reduce the number of inverses you need to compute, so basically a 'combo' is a group of columns that share the same structure.

**Review by jbruestle — COMMENTED** [review]
Responded to some TODOs.

**Comment by BoltonBailey** [comment]
Thanks Jeremy, your comments have been integrated!

**Pull request merged**